packages/Operators/test/tstReferenceHexLocalMap.cpp

```diff
@@ -0,0 +1,173 @@
+TEUCHOS_UNIT_TEST( ReferenceHexLocalMap, local_map_test )
+{
+    // Create the nodes;
+    int num_nodes = 8;
+    Teuchos::Array<DataTransferKit::Entity> nodes( num_nodes );
+    nodes[0] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 0, 0.0, 0.0, 0.0 );
+    nodes[1] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 1, 2.0, 0.0, 0.0 );
+    nodes[2] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 2, 2.0, 2.0, 0.0 );
+    nodes[3] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 3, 0.0, 2.0, 0.0 );
+    nodes[4] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 4, 0.0, 0.0, 2.0 );
+    nodes[5] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 5, 2.0, 0.0, 2.0 );
+    nodes[6] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 6, 2.0, 2.0, 2.0 );
+    nodes[7] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 7, 0.0, 2.0, 2.0 );
+    
+    // Make a hex.
+    DataTransferKit::Entity hex = 
+        DataTransferKit::UnitTest::ReferenceHex( 0, 0, nodes );
+    
+    // Create a local map from the bulk data.
+    Teuchos::RCP<DataTransferKit::EntityLocalMap> local_map =
+	Teuchos::rcp( new DataTransferKit::UnitTest::ReferenceHexLocalMap() );
+
+    // Test the measure.
+    TEST_EQUALITY( local_map->measure(hex), 8.0 );
+    TEST_EQUALITY( local_map->measure(nodes[0]), 0.0 );
+
+    // Test the centroid.
+    Teuchos::Array<double> centroid( 3, 0.0 );
+    local_map->centroid( hex, centroid() );
+    TEST_EQUALITY( centroid[0], 1.0 );
+    TEST_EQUALITY( centroid[1], 1.0 );
+    TEST_EQUALITY( centroid[2], 1.0 );
+    local_map->centroid( nodes[0], centroid() );
+    TEST_EQUALITY( centroid[0], 0.0 );
+    TEST_EQUALITY( centroid[1], 0.0 );
+    TEST_EQUALITY( centroid[2], 0.0 );    
+    local_map->centroid( nodes[1], centroid() );
+    TEST_EQUALITY( centroid[0], 2.0 );
+    TEST_EQUALITY( centroid[1], 0.0 );
+    TEST_EQUALITY( centroid[2], 0.0 );    
+    local_map->centroid( nodes[2], centroid() );
+    TEST_EQUALITY( centroid[0], 2.0 );
+    TEST_EQUALITY( centroid[1], 2.0 );
+    TEST_EQUALITY( centroid[2], 0.0 );
+    local_map->centroid( nodes[3], centroid() );
+    TEST_EQUALITY( centroid[0], 0.0 );
+    TEST_EQUALITY( centroid[1], 2.0 );
+    TEST_EQUALITY( centroid[2], 0.0 );        
+    local_map->centroid( nodes[4], centroid() );
+    TEST_EQUALITY( centroid[0], 0.0 );
+    TEST_EQUALITY( centroid[1], 0.0 );
+    TEST_EQUALITY( centroid[2], 2.0 );    
+    local_map->centroid( nodes[5], centroid() );
+    TEST_EQUALITY( centroid[0], 2.0 );
+    TEST_EQUALITY( centroid[1], 0.0 );
+    TEST_EQUALITY( centroid[2], 2.0 );    
+    local_map->centroid( nodes[6], centroid() );
+    TEST_EQUALITY( centroid[0], 2.0 );
+    TEST_EQUALITY( centroid[1], 2.0 );
+    TEST_EQUALITY( centroid[2], 2.0 );    
+    local_map->centroid( nodes[7], centroid() );
+    TEST_EQUALITY( centroid[0], 0.0 );
+    TEST_EQUALITY( centroid[1], 2.0 );
+    TEST_EQUALITY( centroid[2], 2.0 );    
```

Comment:
Not a big deal. It just itches me that we have so many lines of code for not much testing...